### DISCUSSION TITLE
How to solve this problem on github?

### DISCUSSION THREAD

**Discussion by SEOKKAMONI** [vercel/next.js#48956] [Help]
### Summary

After reading the official documentation and completing the settings, I tried to push to the GitHub repository, and the following problem occurred.

```
remote: error: Trace: 7a1bd39c45e2a395467000cbabd7eb2e4718450c97acb1c24d465f4c044c2140
remote: error: See https://gh.io/lfs for more information.
remote: error: File .yarn/unplugged/@next-swc-darwin-arm64-npm-13.3.1-29d0864b6b/node_modules/@next/swc-darwin-arm64/next-swc.darwin-arm64.node is 100.29 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/woowacourse-integrated-service/Woowa-Auth-Client
 ! [remote rejected] main -> main (pre-receive hook declined)
```
I was able to solve this problem with Git Large File Storage, but I really want to know if this is the correct solution. Please let me know if I'm setting something wrong or if there is an internal problem with Nextjs.

### Additional information

```js
➜ git:(main) git push origin main
Enumerating objects: 147, done.
Counting objects: 100% (147/147), done.
Delta compression using up to 8 threads
Compressing objects: 100% (142/142), done.
Writing objects: 100% (147/147), 94.39 MiB | 7.10 MiB/s, done.
Total 147 (delta 4), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (4/4), done.
remote: error: Trace: 7a1bd39c45e2a395467000cbabd7eb2e4718450c97acb1c24d465f4c044c2140
remote: error: See https://gh.io/lfs for more information.
remote: error: File .yarn/unplugged/@next-swc-darwin-arm64-npm-13.3.1-29d0864b6b/node_modules/@next/swc-darwin-arm64/next-swc.darwin-arm64.node is 100.29 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/woowacourse-integrated-service/Woowa-Auth-Client
 ! [remote rejected] main -> main (pre-receive hook declined)
```

**Comment by icyJoseph** [comment]
What yarn version are you using? It looks like the unplugged folder should be ignored by git (you need to place it in the `.gitignore` file) - https://github.com/yarnpkg/berry/issues/454#issuecomment-530312089

**Reply by SEOKKAMONI** [reply]
I'm using version 3.4.1 of yarn. I also added what you said to gitignore, but the error isn't resolved.

gitingnore
```
# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.

# dependencies
/node_modules
/.pnp
.pnp.js

# testing
/coverage

# next.js
/.next/
/out/

# production
/build

# misc
.DS_Store
*.pem

# debug
npm-debug.log*
yarn-debug.log*
yarn-error.log*
.yarn/*
!.yarn/releases
!.yarn/plugins
.pnp.*

# local env files
.env*.local

# vercel
.vercel

# typescript
*.tsbuildinfo
next-env.d.ts
```

**Reply by icyJoseph** [reply]
yeah but like, you gotta clean those big files from the git history... I think this is why some people stick to yarn 1.x.x (like me) - if you don't mind, delete the `.git` folder, and start over, you'll lose all of the git history up to that point. You could also try to reset back to the commit that added the unwanted .yarn folder, making sure to commit the new `.gitignore` file first.

This is a yarn, git and github issue anyway, nothing Next.js can truly help with, though perhaps they should've ignored those files in the template.

**Reply by wwwdepot** [reply]
Follow this [docs](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#removing-files-from-a-repositorys-history). It includes instructions on how to remove files from a repository's history, including files that were added in earlier commits.

**Reply by nikkuv** [reply]
I tried everything from above but nothing worked for me. Then I deleted my .git file and initialize git again. It worked for me.

**Reply by IllIdanIllI** [reply]
Maybe it will be helpful for the people who don't want to remove the `.git` folder 
As @icyJoseph said, I found that I committed these files for some reason in the first commits
![Screenshot 2024-04-14 at 18 40 35](https://github.com/vercel/next.js/assets/38493704/93ff5301-67d1-42c3-b0b0-8002b3bed2e1)
I suppose because files were removed lately the `git rm --cached` didn't work for me
To reach these commits I used `git rebase -i HEAD~<long commits ago>` and then edited the required commits.
The drawback of this method is that it will override all `<long commits ago> `amount of commits

**Reply by ireneeming** [reply]
how did you fix this problem?

**Reply by bosconian-dynamics** [reply]
**EDIT:** The correction to the `.gitignore` files should be a part of the `v14.3.0` release. If you are comfortable with the potential instability of using pre-releases, you can create a new project with the fix now via `yarn create next-app@v14.3.0-canary.85` (or any later `v14.3.0-canary.XX` pre-release).

----------------------------------

@ireneeming

1. Update your `.gitignore` file's entries for Yarn to match their [official recommendations](https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored); specifically, replace the `.pnp.js` and `.yarn/install-state.gz` lines with the following:  
    ```
    .pnp.*
    .yarn/*
    !.yarn/patches
    !.yarn/plugins
    !.yarn/releases
    !.yarn/versions
    ```  
    (This is a baseline which maintains Yarn's general PnP functionality, but you may need to modify it for your preferences if your project should not leverage PnP or is intended to work as a "Zero-Install")
2. Now you must scrub the `.yarn/unplugged` directory's inclusion from the repository history. There are many different ways to do this, including:  
   - If you don't have any local commit history which you'd like to maintain, you can just delete the `.git` directory and re-initialize the local repo. **This is likely the easiest course of action for a brand new create-next-app project.**
   - If you haven't made any new commits since the one which introduced `./.yarn/unplugged`, you may be able to just `git rm -rf --cached ./.yarn/unplugged`, `git add ./.gitignore`, and `git commit --amend` to remove the directory from the original commit.
   - Stage the removal of the directory with `git rm -rf --cached ./.yarn/unplugged`, add your `.gitignore` modifications, make a new commit, then use a [soft reset](https://stackoverflow.com/a/5201642) to squash all new commits with the initial commit from create-next-app together. This will result in the loss of all intermediary commits from the repo history but will maintain their changes.
   - Use `git filter-branch --index-filter 'git rm -rf --cached --ignore-unmatch ./.yarn/unplugged' HEAD`. This comes with caveats of possibly creating a large number of new commits and mangling history - other options mitigate these issues.
   - Use [git-filter-repo](https://github.com/newren/git-filter-repo).
   - Use [BFG Repo-Cleaner](https://rtyley.github.io/bfg-repo-cleaner).
   - Use `git obliterate ./.yarn/unplugged` from [git-extras](https://github.com/tj/git-extras) (this relies on `git filter-branch` under the hood and comes with the same caveats).
3. If you already have history on a remote like GitHub which already includes the commit introducing the `./.yarn/unplugged` directory, you may need to force-push your repo to overwrite it's history. Note that this will cause any other collaborators' local repositories' history to be disparate and incompatible with the remote - they will need to pull from the remote again to work off of the modified history.

#65586 is tracking this issue. If and when it's merged, #65823 will correct the problem for new CNA projects.
